t/range_query.t

```diff
@@ -0,0 +1,211 @@
+my $term_query_species = Lucy::Search::TermQuery->new(
+    field => 'taxon_id',
+    term  => $taxon_id, 
+);
+
+#get total count of documents
+my $hits_species = $searcher->hits(
+        query      => $term_query_species,
+        num_wanted => 100000000000000000,
+    );
+
+my $hit_count_species = $hits_species->total_hits;  # get the hit count here
+print "======Total doc Count from species =======: $hit_count_species\n";
+ok($hit_count_species == 197782, "Got the right hit count for species $taxon_id");
+
+#============
+
+
+my $term_query_chromosome = Lucy::Search::TermQuery->new(
+    field => 'chromosome',
+    term  => $chromosome, 
+);
+
+
+my $term_query_strand = Lucy::Search::TermQuery->new(
+    field => 'strand',
+    term  => $strand, 
+);
+
+my $taxon_and_chr_strand_query = Lucy::Search::ANDQuery->new(
+    children => [ $term_query_species, $term_query_chromosome, $term_query_strand ],
+);
+
+my $hits_taxon_and_chr_strand = $searcher->hits(
+        query      => $taxon_and_chr_strand_query,
+        num_wanted => 100000000000000000,
+    );
+
+my $hits_taxon_and_chr_strand_count = $hits_taxon_and_chr_strand->total_hits;  # get the hit count here
+print "======Total doc Count from taxon and chr and strand query =======: $hits_taxon_and_chr_strand_count\n";
+
+
+my $range_query1 = Lucy::Search::RangeQuery->new(
+    field         => 'transcript_start', 
+    lower_term    => $transcript_start,
+    upper_term    => $transcript_end,
+    include_upper => 1,
+    include_lower => 1,
+);
+
+
+
+
+my $range_query2 = Lucy::Search::RangeQuery->new(
+    field         => 'transcript_end', 
+    lower_term    => $transcript_start,
+    upper_term    => $transcript_end,
+    include_upper => 1,
+    include_lower => 1,
+);
+
+#lower_term - Lower delimiter. If not supplied, all values less than upper_term will pass.
+my $range_query3 = Lucy::Search::RangeQuery->new(
+    field         => 'transcript_start', 
+    upper_term    => $transcript_start,
+    include_upper => 1,
+);
+
+#upper_term - Upper delimiter. If not supplied, all values greater than lower_term will pass.
+my $range_query4 = Lucy::Search::RangeQuery->new(
+    field         => 'transcript_end', 
+    lower_term    => $transcript_end,
+    include_lower => 1,
+);
+
+#(txStart <= 10000 AND txEnd >= 90500))
+my $and_query_range = Lucy::Search::ANDQuery->new(
+    children => [ $range_query3, $range_query4 ],
+);
+
+#== ((txStart BETWEEN 10000 AND 90500) OR (txEnd BETWEEN 10000 AND 90500) OR (txStart <= 10000 AND txEnd >= 90500))
+my $or_query = Lucy::Search::ORQuery->new(
+    children => [ $range_query1, $range_query2, $and_query_range ],
+);
+
+my $final_query = Lucy::Search::ANDQuery->new(
+    children => [ $taxon_and_chr_strand_query, $or_query ],
+);
+
+
+$hits = $searcher->hits(
+        query      => $final_query,
+        num_wanted => 1000000,
+    );
+
+$hit_count = $hits->total_hits;  # get the hit count here
+print "======HIT Count Final =======: $hit_count\n";
```

Comment:
Very chatty test output, can use note() instead so that you only get the output outside of prove or Travis etc.